examples/go-tc-counter/main.go

```diff
@@ -5,61 +5,32 @@ package main
 
 import (
 	"context"
-	"crypto/tls"
-	"crypto/x509"
-	"flag"
 	"fmt"
 	"log"
 	"os"
 	"os/signal"
-	"strconv"
 	"syscall"
 	"time"
 
 	"github.com/cilium/ebpf"
-	toml "github.com/pelletier/go-toml"
 	gobpfd "github.com/redhat-et/bpfd/clients/gobpfd/v1"
+	"github.com/redhat-et/bpfd/examples/pkg/config-mgmt"
+	"github.com/redhat-et/bpfd/examples/pkg/bpfd-app-client"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
 )
 
 type Stats struct {
 	Packets uint64
 	Bytes   uint64
 }
 
-type Tls struct {
-	CaCert string `toml:"ca_cert"`
-	Cert   string `toml:"cert"`
-	Key    string `toml:"key"`
-}
-
-type Config struct {
-	Iface        string `toml:"interface"`
-	Priority     string `toml:"priority"`
-	Direction    string `toml:"direction"`
-	BytecodeUuid string `toml:"bytecode_uuid"`
-	BytecodeLocation string `toml:"bytecode_location"`
-}
-
-type ConfigData struct {
-	Tls    Tls
-	Config Config
-}
-
 const (
 	DefaultConfigPath     = "/etc/bpfd/gocounter.toml"
```

Comment:
Do we want to use the same `DefaultConfigPath` for all examples?  It seems like it might be better to have a file per example type.

Reply:
@anfredette @astoycos Not sure we need to pass "config" through a TOML file any more. I used it with the original K8s deployment to pass config to the userspace container, and that container used either the UUID in the config to find it's bpf program, or used the rest of the config to send a request to bpfd. Now, in k8s deployment, something outside the userspace program creates a bpfProgram CRD Object, The userspace container just uses a fixed bpfProgram name. We could pass a bpfProgram name in if we wanted to, but probably overkill.

We should leave the TOML file reading in place to allow the certs file location to be overwritten and strip out the config portion, but it's not currently used in the k8s deployment, so would only be for running on the host.

Reply:
I was going to say the same when I was reviewing but I thought it may have been serving some explicit purpose for the non-k8s setup, I'd be happy to get rid of it!

Reply:
In progress

Reply:
Done